scripts/west_commands/runners/nrfjprog.py

```diff
@@ -150,68 +163,131 @@ def get_board_snr(self, glob):
+                         '--snr', self.snr])
 
+    def program_hex(self):
+        # Get the nrfjprog command use to actually program self.hex_.
         self.logger.info('Flashing file: {}'.format(self.hex_))
+
+        # What type of erase argument should we pass to nrfjprog?
         if self.erase:
-            commands.extend([
-                ['nrfjprog',
-                 '--eraseall',
-                 '-f', self.family,
-                 '--snr', board_snr],
-                program_cmd
-            ])
+            erase_arg = '--chiperase'
         else:
-            if self.family == 'NRF51':
-                commands.append(program_cmd + ['--sectorerase'])
-            elif self.family == 'NRF52':
-                commands.append(program_cmd + ['--sectoranduicrerase'])
+            if self.family == 'NRF52':
+                erase_arg = '--sectoranduicrerase'
             else:
-                uicr = {
-                    'NRF53': ((0x00FF8000, 0x00FF8800),
-                              (0x01FF8000, 0x01FF8800)),
-                    'NRF91': ((0x00FF8000, 0x00FF8800),),
-                }[self.family]
-
-                if not self.force and has_region(uicr, self.hex_):
-                    # Hex file has UICR contents.
-                    raise RuntimeError(
-                        'The hex file contains data placed in the UICR, which '
-                        'needs a full erase before reprogramming. Run west '
-                        'flash again with --force or --erase.')
-                else:
-                    commands.append(program_cmd + ['--sectorerase'])
+                erase_arg = '--sectorerase'
 
+        if self.family == 'NRF53':
+            if self.build_conf.get('CONFIG_SOC_NRF5340_CPUAPP', False):
+                coprocessor = 'CP_APPLICATION'
+            elif self.build_conf.get('CONFIG_SOC_NRF5340_CPUNET', False):
+                coprocessor = 'CP_NETWORK'
+            else:
+                # When it's time to update this file, it would probably be best
+                # to handle this by adding common 'SOC_NRF53X_CPUAPP'
+                # and 'SOC_NRF53X_CPUNET' options, so we don't have to
+                # maintain a list of SoCs in this file too.
+                raise RuntimeError(f'unknown nRF53; update {__file__}')
+            coprocessor_args = ['--coprocessor', coprocessor]
+        else:
+            coprocessor_args = []
+
+        # It's important for tool_opt to come last, so it can override
+        # any options that we set here.
+        try:
+            self.check_call(['nrfjprog', '--program', self.hex_, erase_arg,
+                             '-f', self.family, '--snr', self.snr] +
+                            coprocessor_args + self.tool_opt)
+        except subprocess.CalledProcessError as cpe:
+            if cpe.returncode == UnavailableOperationBecauseProtectionError:
+                self.logger.error('Flashing failed because the target '
```

Comment:
Can we warn the user that `west flash --recover` will erase all flashes for all cores?

Reply:
See my comment above, but I would add a warning here too in the case of nRF53, since the user might not read the help for `--recover`